bin/amphtml-update.py

```diff
@@ -772,6 +781,39 @@ def UnicodeEscape(string):
 	"""
 	return ('' + string).encode('unicode-escape')
 
+def GetMandatoryOf( attr, constraint ):
+	"""Gets the mandatory_*of value, if it exists.
+
+	Args:
+		attr: The attributes in which to look for the mandatory_*of constraint.
+		constraint: A string of the mandatory_*of constraint, like 'mandatory_anyof'.
+	Returns:
+		A dictionary with the constraint name mapped to its value.
+	"""
+	for attr_spec in attr:
+		if attr_spec.HasField(constraint):
+			spec_dict = {}
+			mandatory_of = getattr(attr_spec, constraint).strip('[]').split(',')
+			spec_dict[constraint] = [GetMandatoryOfAttr(oneof) for oneof in mandatory_of]
+			return spec_dict
```

Comment:
I have a suggestion to improve the logic here which would eliminate the need for `GetMandatoryOfAttr`. Given that the value is not actually parsed, what can be done is merely to check if the key exists for a given attribute, and of so, add the attribute name to the list for the constraint. 

So this could be replaced with:

```python
attributes = []
for attr_spec in attr:
	if attr_spec.HasField(constraint):
		attributes.append( attr_spec.name )
if len( attributes ) == 0:
	return None
return attributes
```

And then `GetMandatoryOfAttr` could be removed. 

Is that right?

Reply:
Thanks, that's a great idea. 

https://github.com/ampproject/amp-wp/pull/4285/commits/3b1dbf8a830e75d9ac55ffecaecad25124ab40d8 applies it.